### ISSUE TITLE
Automatic synch without requiring to actually "store" the .ipynb notebook

### ISSUE THREAD

**Issue by fschiffers** [fmilanese-1/ds_utils#14]
Hello,

I haven't used your tool yet, but I'm having high hopes on it!

For the last 2 years, I've been using the "Jupytext for Notebooks" by Don Jayamane. Unfortunately, this addon is no longer supported and it's quite buggy.

I hope that at some point, your addon will be a replacement for Jupytext. I'd love to open up a python script directly as a notebook, and any changes in the notebook should be directly synched back into the python script!

Anyways, thanks for putting the effort into this add-on! I'm looking forward to any changes you'll make.

Best,
Florian

**Comment by fmilanese-1** [comment]
Hi thanks!, yes I was on the same situation, that's why I decided to go this way.

You mean something like keeping the actual notebook hidden from the user?... I think that in order to have full integrability with vscode jupyter extensions we need to open a file in the JSON notebook format.

The idea I have is to actually sote and work on a ipynb file, but under the hood sync it with the percent script version of jupytext, that way you keep all vscode jupyter features, and all jupytext transcription.

If you are working with git, then you can simply gitignore any .ipynb and keep track only of .py scripts.

**Comment by fschiffers** [comment]
Have you ever looked into the jupytext extension?

It's working "okay", but a bit buggy. Anyways, it's somehow creating a .ipynb file without physically saving it ( or at least only temporarily).
There are bugs ( e.g. when you open 2 notebooks at the same time), where the . ipynb file is eventually saved, and the synch is lost.

Anyways: I think if your add-on would mimic the jupytext extension that already exists, that would make it super useful.

**Comment by fmilanese-1** [comment]
Yes, I'm aware how Jayamane's idea works. The current status of this plugin tries to mimic what the jupytext jupyter extension does. Basically, keeping both the script and the jupyter file together and in sync... I think is useful, the user can have control over which version is the "true" version.

And as I say, if you work with git, you can simply gitignore the ipynb file and only store the source code.

Maybe we can have this hidden notebook option optional?

**Comment by fschiffers** [comment]
I don't fully understand how Jayamane keeps this "hidden" notebook. All I know is that it is super buggy. I agree with your your idea of keeping .ipynb and .py files in sync, and just adding .ipynb to gitignore might be the better solution! 
I'm working on a framework and a few users said they find "jupytext" too complicated and want to use .ipynb files directly, as there's less friction to execute actual code. I guess ocassionaly saving a "clear" .ipynb (and synched to the .py file) without all the meta data might be the way to go here. 

For now, I'll wait for your automatic synch option between .ipynb and .py. I think the worst that could happen is that someone uses an .ipynb notebook, fails to synch it to the .py file and then somehow "synchs from .py to .ipynb) which would delete all the work someone has done in the .ipynb itself.

**Comment by fmilanese-1** [comment]
If you have version control and need to peer review, peer review json notebooks is humanly impossible. I think that is the core idea why jupytext exists.

The automatic sync is there, you just need to save the ipynb with cntrl+shift+s and it save and syncs it.

**Comment by fschiffers** [comment]
Thanks, I'll give it a try later. I missed the auto sync as I thought this would be a future option.

When I encounter bugs, I'll add issues here to help you improve the app 😄

Thanks for the quick feedback!
